### PR TITLE
Mode tiff

### PR DESCRIPTION
**Pull request by tmanolis**
`Refacto / Architecture` : 
- put all code files for flags in a directory and create a file .cpp per flag
- add 2 new variables in the `Channel Class` :
```cpp
std::vector<std::string>	_banned_users;
std::vector<std::string> _voiced_users;
int _capacity_limit;
```

Add 4 new flags for the commande `/MODE` :
- `BAN (+b)` :  
This channel mode controls a list of client masks that are ‘banned’ from joining or speaking in the channel.

If this mode is set on a channel, and a client sends a JOIN request for this channel, their nickmask  is compared with each banned client mask set with this mode. If they match one of these banned masks, they will receive an [ERR_BANNEDFROMCHAN](474) reply and the JOIN command will fail.

- `LIMIT (+l)` : 
This channel mode controls whether new users may join based on the number of users who already exist in the channel. If this mode is set, its value is an integer and defines the limit of how many clients may be joined to the channel.

If this mode is set on a channel, and the number of users joined to that channel matches or exceeds the value of this mode, new users cannot join that channel. If a client sends a JOIN request for this channel, they will receive an [ERR_CHANNELISFULL](471) reply and the command will fail.

- `MODERATE (+m)` :
This channel mode controls whether users may freely talk on the channel, and does not have any value.

If this mode is set on a channel, only users who have channel privileges may send messages to that channel. The [voice] channel mode is designed to let a user talk in a moderated channel without giving them other channel moderation abilities, and users of higher privileges may also speak in moderated channels.

- VOICE Prefix (+v)
This mode is standard. The prefix and mode letter used for it, respectively, are "+" and "+v".
Users with this mode may send messages to a channel that is [moderated].

### CODE REVIEW

File: srcs/commands/join.cpp [hunk 1]
```diff
@@ -70,17 +70,25 @@ void	join(Server *server, int const client_fd, cmd_struct cmd_infos)
...
 		}
 
-		// vérifier si le client est banned avant de le join au channel
 		std::map<std::string, Channel>::iterator it_chan = server->getChannels().find(channel_name);
-		if (it_chan->second.isBanned(client_nickname) == true) {
+		// vérifier si le channel est full (METTRE UN PUTAIN DE RETURN avec l'error)
+		if (it_chan->second.getCapacityLimit() != -1 && (int)it_chan->second.getClientList().size() == it_chan->second.getCapacityLimit())
+		{
+			addToClientBuffer(server, client_fd, ERR_CHANNELISFULL(client_nickname, channel_name));
+			return ;
```
Comment (marineks) [inline srcs/commands/join.cpp]: Est-ce que ca vous va si on met un `continue` à la place ? Car comme c'est une boucle, dans le cas où l'on a la commande `/JOIN chan1,chan2 <key1,key2>` il faut que l'on puisse checker son accessiblité au chan2 s'il est ban du chan1

Comment (marineks) [reply srcs/commands/join.cpp]: Et très beau travail de modes ! C'est ouf tous les cas qui sont gérés maintenant :)

Comment (tmanolis) [inline srcs/commands/join.cpp:83]:
```suggestion
			continue ;
```
will be modify like that

Comment (tmanolis) [reply srcs/commands/join.cpp]: Will be modify as @marineks suggested :
```suggestion
			continue ;
```

**Pull request merged**